File: palindrome.go
func isPalindrome(x int) bool {
    if x < 0{
        return false
    } else if x == 0 {
        return true
    }
    
    var  arr  [12] int;
    var arrSize=0;
    var v = x;
    for{
        if v <= 0 {break} 
        var digit = v % 10;
        arr[arrSize]=(digit);
        arrSize+=1
        v /= 10;
    }
    
    for i,j := 0, arrSize-1; i < j ; i,j = i+1, j-1{
        if arr[i] != arr[j]{
             return false
        }            
    }
    
    return true
}

File: palindrome.c


bool isPalindrome(int x){
if (x < 0) return false;
        else if (x ==0) return true;
        else {
            int  arr[1000];
            int arrSize=0;
            int v = x;
            int digit;
            while(v > 0) {
                 digit = v % 10;
                arr[arrSize++]=(digit);
                v /= 10;
            }
            int i=0;
            int j=arrSize-1;
            bool ret = true;
            while(i<j){
               if(arr[i] == arr[j]) {
                   i++;j--;
               } else{
                   ret = false;
                   break;
               }
            }
            return ret;
        }
}
